src/zst_decompressor.ts

```diff
@@ -232,28 +234,39 @@ const decompressAndAnalyze = async (file, start = 0) => {
           });
 
           result.on('end', async () => {
+            // create chunks of size 10 from filesProduced
+            const filesProducedArray = Array.from(filesProduced);
+            const fileGroups: string[][] = [];
+            for (let i = 0; i < filesProducedArray.length; i += 10) {
+              fileGroups.push(filesProducedArray.slice(i, i + 10));
+            }
+            console.log(`file groups: ${fileGroups.length}`);
+
             // When all data is decompressed, run the analysis on the produced files concurrently
-            for (const file of Array.from(filesProduced).slice(0, 10)) {
+            for (const group of fileGroups) {
+              console.log(
+                `Running analysis on ${group.length} files... in group`
+              );
+              // TODO: This is debug code, I believe
               // TODO: this won't work out of the box for a large number of files as there is no max concurrency. But the sample only produces 4 decompressed files
               // I'm slicing to test this with a smaller number of files
 
-              analysisPromises.push(runAnalysis(file));
-              filesBeingAnalyzed.add(newFilePath);
-            }
-
-            // When all analyses are done, delete the files from the set
-            Promise.allSettled(analysisPromises)
-              .then(() => {
+              const analysisPromises = group.map((file) => runAnalysis(file));
+              // When all analyses are done, delete the files from the set
+              try {
+                console.log('Awaiting all analyses to complete...');
+                await Promise.all(analysisPromises);
                 console.log('All analyses completed');
-                for (const file of filesBeingAnalyzed) {
+                for (const file of group) {
                   if (fs.existsSync(file)) {
                     fs.unlinkSync(file);
                     console.log(`File ${file} has been deleted.`);
                   }
                 }
-                filesBeingAnalyzed.clear();
-              })
-              .catch(console.error);
+              } catch (error) {
+                console.error(`Error running analysis: ${error.message}`);
+              }
```

Comment:
@bennyrubanov chgs

Reply:
apparently, your rewrite will not delete the file in a given promise if promise.all rejects a promise:

Error Handling:
In the first chunk, errors in individual promises do not prevent the execution of code that deletes the files. Promise.allSettled ensures that the deletion logic runs even if some analyses fail.
In the second chunk, if any promise rejects, the error is caught, and the file deletion logic is not executed. This is because the rejection of any promise in Promise.all causes it to reject immediately, skipping the rest of the try block.


This is okay with me, as long as that's what we want to happen. I think I'd probably prefer it to delete regardless so we dont end up with a bunch of trash

Reply:
do not follow what is different in this impl vs the previous as far as file deletion

Reply:
allSettled vs all?

Reply:
looked it up and yes

Reply:
will use allSettled